weather_mv/loader_pipeline/bq.py

```diff
@@ -305,6 +328,17 @@ def extract_rows(uri: str,
 
     with open_dataset(uri, open_dataset_kwargs, disable_grib_schema_normalization,
                       tif_metadata_for_datetime) as ds:
+
+        # Find the grid_resolution. In case of single point we can't find grid resolution.
+        should_create_polygon = False
+        if ds['latitude'].size > 1 and ds['longitude'].size > 1:
+            # consider that Grid is regular.
```

Comment:
Maybe, launch a warning log message with this? Related: in the BQ constructor, let's add a check on the test data point to ensure that the grid is regular. If it isn't, let's omit adding the polygon.

Reply:
Last thought on this: In the constructor, if we can't create polygons, let's add a warning log message.